test/unittest/tests/block/test_block_other.cpp

```diff
@@ -0,0 +1,125 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_other]: block.h
+ * This file unit tests the __wt_block_header_byteswap_copy and __wt_block_eligible_for_sweep
+ * functions.
+ */
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+
+static void
+test_block_header_byteswap_copy(WT_BLOCK_HEADER *from, WT_BLOCK_HEADER *to)
+{
+    WT_BLOCK_HEADER prev_from, prev_to;
+
+    // Save the original values before any potential byte re-orderings.
+    prev_from.disk_size = from->disk_size;
+    prev_from.checksum = from->checksum;
+    prev_to.disk_size = to->disk_size;
+    prev_to.checksum = to->checksum;
+
+#ifdef WORDS_BIGENDIAN
+    __wt_block_header_byteswap_copy(from, to);
+    REQUIRE(to->checksum == __wt_bswap32(prev_from.checksum));
+    REQUIRE(to->disk_size == __wt_bswap32(prev_from.disk_size));
+#else
+    // Test that the byte orderings remain the same in both block headers.
+    REQUIRE(from->disk_size == prev_from.disk_size);
+    REQUIRE(from->checksum == prev_from.checksum);
+    REQUIRE(to->disk_size == prev_to.disk_size);
+    REQUIRE(to->checksum == prev_to.checksum);
+#endif
+}
+
+TEST_CASE("Block header byteswap copy", "[block_other]")
+{
+    WT_BLOCK_HEADER from, to;
+
+    SECTION("Test case 1")
+    {
+        from.disk_size = 12121;
+        from.checksum = 24358;
```

Comment:
I think it would be good to include at least one test case where every `checksum` byte is non-zero, as real checksums will rarely include zero bytes.

It is wildly unlikely to happen in the real world, but an implementation that (incorrectly) always sets the lowest byte to zero when doing the swap would pass these tests.

I suppose the same argument applies to `disk_size`, although the reality is that we're not going to have block sizes that big.